public/pages/Rules/store/RulesStore.ts

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { RuleService } from '../../../services';
+import { load, safeDump } from 'js-yaml';
+import { RuleItemInfoBase, IRulesStore, IRulesCache } from '../../../../types';
+import { Rule } from '../../../../models/interfaces';
+import { NotificationsStart } from 'opensearch-dashboards/public';
+import { errorNotificationToast } from '../../../utils/helpers';
+import { ruleTypes } from '../utils/constants';
+import _ from 'lodash';
+
+/**
+ * Class is used to make rule's API calls and cache the rules.
+ * If there is a cache data requests are skipped and result is returned from the cache.
+ * If cache is invalidated then the request is made to get a new set of data.
+ *
+ * @class RulesStore
+ * @implements IRulesStore
+ * @param {BrowserServices} services Uses services to make API requests
+ */
+export class RulesStore implements IRulesStore {
+  /**
+   * Rule service instance
+   *
+   * @property {RuleService} service
+   * @readonly
+   */
+  readonly service: RuleService;
+
+  /**
+   * Notifications
+   * @property {NotificationsStart}
+   * @readonly
+   */
+  readonly notifications: NotificationsStart;
+
+  constructor(service: RuleService, notifications: NotificationsStart) {
+    this.service = service;
+    this.notifications = notifications;
+  }
+
+  /**
+   * Keeps rule's data cached
+   *
+   * @property {IRulesCache} cache
+   */
+  private cache: IRulesCache = {};
+
+  /**
+   * Invalidates all rules data
+   */
+  private invalidateCache = () => {
+    this.cache = {};
+    return this;
+  };
+
+  /**
+   * Returns all rules, custom and pre-packaged
+   *
+   * @method getAllRules
+   * @param {terms?: { [key: string]: string[] }} [terms]
+   * @returns {Promise<RuleItemInfoBase[]>}
+   */
+  public async getAllRules(terms?: { [key: string]: string[] }): Promise<RuleItemInfoBase[]> {
+    let prePackagedRules = await this.getRules(false, terms);
```

Comment:
why are we passing `false` for `prePackaged` and `true` for `custom`?

Reply:
prePackaged is pat of the request query, I could add two more methods, getPackagedRules, and getCustomRules which would have these params already set, just to be more convenient.